pkg/operator/target_status.go

```diff
@@ -362,29 +313,29 @@ func updateTargetStatus(ctx context.Context, logger logr.Logger, kubeClient clie
 		return err
 	}
 
-	var patchErr error
+	var errs []error
 	for job, endpointStatuses := range endpointMap {
-		// Kubelet scraping is configured through hard-coding and not through
-		// a PodMonitoring. As there's no status to update, we skip.
-		if strings.HasPrefix(job, "kubelet") {
```

Comment:
IIUC the bug is we try and build the PodMonitoring to patch if we encounter a NodeMonitoring job, correct?

If so, couldn't the fix be a lot smaller with a simple
```go
if !(strings.HasPrefix("PodMonitoring/") || strings.HasPrefix("ClusterPodMonitoring")) {
	continue
}
```

with some new test cases to validate?

Reply:
I updated the original PR description.

tl;dr yes it could be a lot smaller. I pulled in some changes from https://github.com/GoogleCloudPlatform/prometheus-engine/pull/766 because I felt like it made the code cleaner overall.

My personal opinion is that simple is best if we want the fix out quickly. I confirmed with @macxamin that NodeMonitoring won't be used for a while so we we can just include this change as part of our next (hopefully biweekly) release. @macxamin feel free to let me know if I'm wrong or if something changes.

Reply:
Yea, I did what @pintohutch suggested in other PR to get unblocked, but I'm happy for @TheSpiritXIII to fix this properly here 👍🏽  (and add support). We can always port or quickly add fix to the release if needed... 

Let's decide if we want 0.10 to have node monitoring properly, we can announce it on 0.11 or do 0.10.1 up to us

Reply:
Gotcha.

Thanks for the context. The new PR subscription colors this in nicely 😄 .